src/main/java/edu/RobertoAranovich/advancedjava/StockServiceFactory.java

```diff
@@ -0,0 +1,25 @@
+package edu.RobertoAranovich.advancedjava;
+
+/**
+ * Implementation of StockServiceFactory class.
+ *
+ * @author Roberto Aranovich
+ */
+
+public class StockServiceFactory {
+
+    /**
+     * Implementation of getQuote method that returns a StockQuote with a hard coded price value.
+     *
+     * @param type an element from the StockServiceType enumeration.
+     */

```

Comment:
the idea behind the factory is that the client code (the code calling the factory)  does not know anything about the implement details (the concrete class). It just knows it is getting back something that satisfies the interface. 
In this case, though, the client is telling the factory what implementation it wants. This is not correct. I know that many online examples of the factory pattern so it working the way you have it, but doing it this way defeats some of the goals we are trying to archive. 
1) client code does know about implementing. 
2) functionality can be easily changed by just changing the factory class and nothing else (this is huge).

Reply:
I see...

Reply:
Does it make sense to return different implementations of StockService depending on what arguments are passed to the factory? For instance, if no arguments are passed, return BasicStockService, if a database is passed, return a StockService that makes use of that database?